client/components/data/activating-theme/index.js

```diff
@@ -1,49 +1,49 @@
 /**
  * External dependencies
  */
-var React = require( 'react' );
+import React from 'react';
+import { connect } from 'react-redux';
+import omit from 'lodash/object/omit';
 
 /**
  * Internal dependencies
  */
-var CurrentThemeStore = require( 'lib/themes/stores/current-theme' );
-
-function getState( props ) {
-	return {
-		isActivating: CurrentThemeStore.isActivating(),
-		hasActivated: CurrentThemeStore.hasActivated(),
-		currentTheme: CurrentThemeStore.getCurrentTheme( props.siteId )
-	};
-}
+import {
+	isActivating,
+	hasActivated,
+	getCurrentTheme
+} from 'lib/themes/selectors/current-theme';
 
 /**
  * Passes the activating state of themes to the supplied child component.
  */
-var ActivatingThemeData = React.createClass( {
+const ActivatingThemeData = React.createClass( {
 
 	propTypes: {
-		children: React.PropTypes.element.isRequired
-	},
-
-	getInitialState: function() {
-		return getState( this.props );
-	},
-
-	componentWillMount: function() {
-		CurrentThemeStore.on( 'change', this.onActivatingTheme );
-	},
-
-	componentWillUnmount: function() {
-		CurrentThemeStore.off( 'change', this.onActivatingTheme );
-	},
-
-	onActivatingTheme: function() {
-		this.setState( getState( this.props ) );
+		children: React.PropTypes.element.isRequired,
+		// The following props aren't part of this module's public interface,
+		// since they are provided by the `connect()` call that is wrapped around
+		// `ActivatingThemeData` before it is exported.
```

Comment:
> Hmm, how would we achieve the latter? I can seem to think only of ES6 class/propTypes-style, i.e. adding a `propTypes` member after it's connected...

Yeah, that was my rough idea. Inspection via web tools shows that a connected component automatically has `propTypes: { store }` attached to it. On second consideration, I don't know if we should interfere with that mechanism, plus it may not be useful/visible enough.

I'd lean towards the pseudo-JSDoc statement, but I'll welcome anything more correct than that.